### PR TITLE
DatastoreCluster DSC Resource

### PR DESCRIPTION
**Pull request by SimeonGerginov**
### Added
- Added **DatastoreCluster** DSC Resource.
- Added Unit Tests for **DatastoreCluster** DSC Resource.
- Added Integration Tests for **DatastoreCluster** DSC Resource.
- Added Documentation and example Configurations for **DatastoreCluster** DSC Resource.

### Changed
- Extended **TestUtils** with helper function for importing the **VMware.vSphereDSC** module.

### CODE REVIEW

**Comment by SimeonGerginov** [comment]
Example usage of DatastoreCluster DSC Resource:

```
DatastoreCluster DatastoreCluster {
    Server = $Server
    Credential = $Credential
    Name = 'DscDatastoreCluster'
    Location = ''
    DatacenterName = 'Datacenter'
    DatacenterLocation = ''
    Ensure = 'Present'
    IOLatencyThresholdMillisecond = 50
    IOLoadBalanceEnabled = $true
    SdrsAutomationLevel = 'FullyAutomated'
    SpaceUtilizationThresholdPercent = 50
}
```

File: Source/VMware.vSphereDSC/DSCResources/DatastoreCluster.ps1 [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+    <#
+    .DESCRIPTION
+
+    Checks if the specified Datastore Cluster configuration should be modified.
+    #>
+    [bool] ShouldModifyDatastoreCluster($datastoreCluster) {
+        $shouldModifyDatastoreCluster = @()
+
+        $shouldModifyDatastoreCluster += ($null -ne $this.IOLatencyThresholdMillisecond -and $this.IOLatencyThresholdMillisecond -ne $datastoreCluster.IOLatencyThresholdMillisecond)
```
Comment (danielboth) [inline Source/VMware.vSphereDSC/DSCResources/DatastoreCluster.ps1:188]: May I suggest using a method to do all these compares? That way you could also tackle the issue described in #264, it also prevents a lot of code duplication. For example that method could look like: 

```PowerShell
[bool] CompareProperty($unsetValue, $intentValue, $currentValue, $propertyName){
    if ($unsetValue -ne $intentValue -and $intentValue -ne $currentValue) {
        Write-Verbose "Setting $propertyName: Current setting [ $currentValue ] does not match desired setting [ $intentValue ]"
        return $false
    } else {
        return $true
    }
}
```

calling it in this function would then be:
```PowerShell
[bool] ShouldModifyDatastoreCluster($datastoreCluster) {
    $shouldModifyDatastoreCluster = @(
        $this.CompareProperty($null, $this.IOLatencyThresholdMillisecond, $datastoreCluster.IOLatencyThresholdMillisecond, 'IOLatencyThresholdMillisecond')
        $this.CompareProperty($null, $this.IOLoadBalanceEnabled, $datastoreCluster.IOLoadBalanceEnabled, 'IOLoadBalanceEnabled')
        $this.CompareProperty([DrsAutomationLevel]::Unset, $this.SdrsAutomationLevel.ToString(), $datastoreCluster.SdrsAutomationLevel.ToString(), 'SdrsAutomationLevel')
        $this.CompareProperty($null, $this.SpaceUtilizationThresholdPercent, $datastoreCluster.SpaceUtilizationThresholdPercent, 'SpaceUtilizationThresholdPercent')
    )

    return ($shouldModifyDatastoreCluster -contains $false)
}
```

**Comment by danielboth** [comment]
Epic you got this so quickly @SimeonGerginov, looking at this, what would be your vision on adding datastores to a datastore cluster through DSC?

File: Source/VMware.vSphereDSC/DSCResources/DatastoreCluster.ps1 (hunk 1 above)
Comment (SimeonGerginov) [reply Source/VMware.vSphereDSC/DSCResources/DatastoreCluster.ps1:188]: Great addition! We can add it to the **BaseDSC** class and then call it in each DSC Resource in the module. I will start working on that now.

**Comment by SimeonGerginov** [comment]
> Epic you got this so quickly @SimeonGerginov, looking at this, what would be your vision on adding datastores to a datastore cluster through DSC?

Well, we have two options that I can think of:

1. Add another property to the current DSC Resource - **DatastoreNames** with which we specify which Datastores should be in the DatastoreCluster.
2. Create a new DSC Resource that is similar to the [VDSwitchVMHost DSC Resource](https://github.com/vmware/dscr-for-vmware/wiki/VDSwitchVMHost) where we specify the DatastoreCluster and the **DatastoreNames** which should be part of the DatastoreCluster.

I prefer the second approach, so that we can be consistent with the previous DSC Resources but also this way the logic for adding/removing Datastores will be separated from the creation and modification of the DatastoreCluster. And this way with the three DSC Resources: **VmfsDatastore**, **DatastoreCluster** and the new one you can create a Configuration that can be managed more easily.

Maybe we should create a separate issue where the discussion can occur ?

**Pull request merged**